src/ape/managers/compilers.py

```diff
@@ -126,9 +134,19 @@ def compile(self, contract_filepaths: List[Path]) -> Dict[str, ContractType]:
             for source_id in source_ids:
                 logger.info(f"Compiling '{source_id}'.")
 
-            compiled_contracts = self.registered_compilers[extension].compile(
-                paths_to_compile, base_path=contracts_folder
-            )
+            compiler = self.registered_compilers[extension]
+
+            original_settings = compiler.compiler_settings
+            if settings is not None:
+                compiler.compiler_settings = settings
+
+            try:
+                compiled_contracts = self.registered_compilers[extension].compile(
```

Comment:
I think it'd be better to update `CompilerAPI.compile` to allow settings override instead of this but I think that is a breaking change, so it may have to wait until 0.7 or 0.8